### PR TITLE
Making the tests build and work with dotnet test command

### PR DESCRIPTION
**Pull request by livarcocc**
Making the test projects build and work with dotnet test. I am not adding them back to the build just yet, but a developer can simply run dotnet test from a test folder and the tests are going to be executed. Everything that depends on System.Slices is failing right now, which is why the tests haven't been re-enabled.

### CODE REVIEW

**Comment by livarcocc** [comment]
@KrzysztofCwalina @sokket @piotrpMSFT PTAL

File: src/System.Buffers/tests/project.json [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
           "version": "0.1.0-*"
         },
     "System.Slices": "0.1.0-*",
-    "xunit": "2.1.0"
+    "xunit": "2.1.0",
+    "dotnet-test-xunit": "1.0.0-dev-45337-57"
   },
   "frameworks": {
-    "dotnet5.6": {
+    "dnxcore50": {
```
Comment (KrzysztofCwalina) [inline src/System.Buffers/tests/project.json]: Why is this change needed?

Comment (livarcocc) [inline src/System.Buffers/tests/project.json]: That's what the tests should be compiled for. Its dependencies won't work for dotnet5.6. We can have dnxcore50 depending on dotnet5.6, but not the other way around.

Comment (KrzysztofCwalina) [inline src/System.Buffers/tests/project.json]: Ah get it. I assume this is temporary though. Once our test dependencies move to std library, we would stop using dnxcore50.

Anyway, the change looks good.

**Comment by KrzysztofCwalina** [comment]
Looks good.
